src/libai/io/MatrixIO.java

```diff
@@ -0,0 +1,214 @@
+        map.put("a", m);
+        
+        write(output, map, t);
+    }
+    
+    /**
+     * Writes a set of {@link Matrix} objects to a given {@link OutputStream}.<br><br>
+     * <i>Note:</i> When saving to {@link Target#CSV} and {@link Target#TSV}, a line separator 
+     * {@literal \n} will be inserted between matrices
+     * <i>Note 2:</i> Since Java doesn't support deserializing multiple objects from the same 
+     * {@link OutputStream} if target is {@link Target#SERIAL} then the whole {@code Map} will
+     * be serialized.
+     * 
+     * @param output The {@link OutputStream} in which to write the {@code Matrix} objects
+     * @param m a {@link Map} where the keys are matrix names, if the target is anything but 
+     * {@link Target#OCTAVE} the name will be ignored
+     * @param t {@link Target}ed format, the default value (if {@code null}) is
+     * {@link Target#SERIAL}
+     * @throws IllegalArgumentException if either {@code output} or {@code m} are {@code null}
+     * @throws IOException if an I/O error occurs
+     */
+    public static void write(OutputStream output,
+                             Map<String, Matrix> m,
+                             Target t) throws IllegalArgumentException, IOException {
+        if (output == null) {
+			throw new IllegalArgumentException("OutputStream can't be null");
+		}
+		if (m == null) {
+			throw new IllegalArgumentException("The matrix map can't be null");
+		}
+		if (m.isEmpty()) {
+            return;
+		}
+        
+        t = t == null ? Target.SERIAL : t;
+        
+        switch(t) {
+            case CSV:    writeText(output, m, ","); break;
+            case TSV:    writeText(output, m, "\t"); break;
+            case OCTAVE: writeOctave(output, m); break;
+            case SERIAL:
+            default:     writeSerial(output, m);
+        }
+    }
+    
+    private static void writeSerial(OutputStream output, Map<String, Matrix> m) throws IOException {
+        ObjectOutputStream oos = new ObjectOutputStream(output);
+        if (m.size() == 1) {
+            oos.writeObject(m.values().iterator().next());
+        } else {
+            oos.writeObject(m);
+        }
+    }
+    
+    private static void writeText(OutputStream output, Map<String, Matrix> m, String sep) throws IOException {
+        PrintStream ps = new PrintStream(output);
+        int k = 0;
+        for (Matrix matrix : m.values()) {
+            for (int i = 0, r = matrix.getRows(); i < r;i++) {
+                for (int j = 0, c = matrix.getColumns(); j < c; j++) {
+                    ps.append(Double.toString(matrix.position(i, j)));
+                    if (j != c - 1) ps.append(sep);
+                }
+                if (i != r - 1) ps.append("\n");
+            }
+            if (++k != m.size()) ps.append("\n");
+        }
+    }
+    
+    private static void writeOctave(OutputStream os, Map<String, Matrix> m) throws IOException {
```

Comment:
Great, I'll read it over the weekend, `Level 5` support will be a nice addition, since `Level 7` is actually `HDF5`.

Level 1 (the current implementation) allows testing using Octave directly without all the mocking about with rounding and copy/pasting.